### ISSUE TITLE
Unconditionally setting Dtr in SerialPort Open fails in some devices that doesn't support Dtr

### ISSUE THREAD

**Issue by maloo** [dotnet/runtime#37841]
### Description

Recreate issue requires a USB serial device that doesn't allow Dtr setting. In this case it was a Sirit/Motorola/Zebra FX9500 UHF RFID reader.
The code to reproduce is just "new SerialStream("COM123", 115_200).Open()"
This will fail with the error "Device is not functioning properly". Error occurs in DtrEnable that gets set in ctor of SerialStream [here](https://github.com/dotnet/runtime/blob/master/src/libraries/System.IO.Ports/src/System/IO/Ports/SerialStream.Windows.cs#L633).
I have downloaded dotnet/runtime and built the System.IO.Ports.dll without [this](https://github.com/dotnet/runtime/blob/master/src/libraries/System.IO.Ports/src/System/IO/Ports/SerialStream.Windows.cs#L183) line. Then everything works fine.

Expected behavior is that DtrEnable is not set unconditionally in the constructor. It should only be set if is different from device current value. If this value is impossible to get, there should be some other way to configure SerialPort before Open that prevent this call from happening. The current behavior is making it impossible to use many devices. Behavior is also not the same in Windows and Linux. In Linux it is possible to get around the issue by setting DtrEnable property of SerialPort before calling Open, this can prevent SerialStream from setting the value. But this setting has to be hard coded to the value that works with the device, so Linux behavior is also broken and should also be fixed.

### Configuration

Same on .Net 3.1 and .Net 5 preview 5
Running on Windows x64, sort of works in Linux

### Regression?

I think it used to work in .Net Framework, but it could have been another type of device. Latest version of nuget does not work. But it has never worked in .Net Core 3.x or 5.0.

@krwq, anything familiar?

**Comment by Jozkee** [comment]
> I have downloaded dotnet/runtime and built the System.IO.Ports.dll without this line. Then everything works fine.

@maloo Doing this doesn't cause any test to fail? 
If this used to work on .NET FW then it may be a functional regression, however it needs more investigation.

**Comment by maloo** [comment]
@Jozkee It was just a POC that it is this code that makes some types of devices fail and that it makes no sense having a constructor setting it for no other reason than "it has always been like that". And that the behavior is inconsistent in Linux and Windows. I'm not sure what solution you would prefer. Adding a property would be a change that require API review I guess. Not setting it if already have same value might be the best option, but I have not tried anything like that because I'm not familiar with this code.
Tests might fail when removing this line, I don't know, because I could not build runtime repo, I had to hack it to get System.IO.dll to build. You can see here: [https://github.com/dotnet/runtime/discussions/37833](https://github.com/dotnet/runtime/discussions/37833).

**Comment by maloo** [comment]
@Jozkee How can we proceed on this? If you don't give me a suggestion of what type of fix (api change or internal hack) I will not even be able to do a pull request. Could you at least tell me what solution you would prefer for a PR of this issue?

**Comment by krwq** [comment]
@maloo are you getting an exception while setting or when using the device after setting? What behavior would you expect?

**Comment by maloo** [comment]
@krwq , sorry, I missed the notification ...
The error is an exception when calling Open on the SerialPort class (probably typo in OP above). SerialPort.Open will create a SerialStream, which will unconditionally set DtrEnable property (value provided to SerialStream constructor by SerialPort). On Windows this will return a Win32 error while setting DtrEnable, because the device does not support it. SerialStream will then raise an error, leading to SerialStream.ctor and SerialPort.Open failing. There is no way I can find to get around this issue in Windows, in Linux DtrEnable set will check current value and if same not set it. But I would prefer that you wouldn't have to know what works for every type of device. DtrEnable should not be set unconditionally. And if you don't want to change that behavior, I'm fine with some property to disable DtrEnable/RtsEnable using some property in SerialPort. Making it bool? would be a breaking change I guess, and I assume not setting it be default might also be a breaking change. I can make a PR, but I would like to know what solution you would prefer first.
Just saw I posted an issue about this before (#27729) and there are more issues on this matter.

**Comment by maloo** [comment]
#31523
#30877

**Comment by krwq** [comment]
@maloo I suggest to proceed with what we do right now except make sure we do not throw in the constructor (i.e. call native API directly but ignore error)

**Comment by jonmill** [comment]
hey folks - I'm hitting this running on a Raspberry PI communicating over USB-to-serial where the `Open` call throws an error since the USB-to-serial device doesn't support DTR, so I can't use the port. This is fully blocking since there is no way to proceed...is there any update here? Thanks! Stack trace below

```
Unhandled exception. System.IO.IOException: Input/output error
   at System.IO.Ports.SerialStream.set_DtrEnable(Boolean value)
   at System.IO.Ports.SerialStream..ctor(String portName, Int32 baudRate, Parity parity, Int32 dataBits, StopBits stopBits, Int32 readTimeout, Int32 writeTimeout, Handshake handshake, Boolean dtrEnable, Boolean rtsEnable, Boolean discardNull, Byte parityReplace)
   at System.IO.Ports.SerialPort.Open()
   at Workout.Equipment.Program.Main(String[] args) in <path>\Program.cs:line 77
   at Workout.Equipment.Program.<Main>(String[] args)
Aborted (core dumped)
```

**Comment by danmoseley** [comment]
@jonmill I expect it has not been investigated. Serial port bugs tend to be expensive to investigate not least because of needing hardware that reproduces it. Are you in a position to debug and investigate a possible fix?

**Comment by jonmill** [comment]
@danmoseley - it looks like there's a PR open that has sign off but hasn't been merged (#48577)...it looks to have Crypto test failures that seem unrelated to the PR (infrastructure issues? flaky test?). What's the protocol for moving forward there...it looks like that PR would solve the issue

**Comment by krwq** [comment]
Hey @jonmill, I've just merged the PR, the failure wasn't related. I was hoping for a second sign-off but given no one did in 2 week period I went forward with the fix. Please let us know once you validate the fix works for you
